pkg/aws/ec2/api/cleanup/node_cleanup.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cleanup
+
+import (
+	"fmt"
+
+	"github.com/aws/amazon-vpc-resource-controller-k8s/pkg/aws/ec2/api"
+	ec2API "github.com/aws/amazon-vpc-resource-controller-k8s/pkg/aws/ec2/api"
+	enicleanup "github.com/aws/amazon-vpc-resource-controller-k8s/pkg/aws/ec2/api/cleanup/eni"
+	"github.com/aws/amazon-vpc-resource-controller-k8s/pkg/config"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/go-logr/logr"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// NodeTerminationCleanerto handle resource cleanup at node termination
+type NodeTerminationCleaner struct {
+	NodeName   string
+	EC2Wrapper ec2API.EC2Wrapper
+	Log        logr.Logger
+}
+
+func (n *NodeTerminationCleaner) GetENITagFilters() []*ec2.Filter {
+	nodeFilter := enicleanup.CommonNetworkInterfaceFilters
+	nodeFilter = append(nodeFilter, []*ec2.Filter{
+		{
+			Name:   aws.String("tag:" + config.NetworkInterfaceNodenameKey),
+			Values: []*string{aws.String(n.NodeName)},
+		},
+	}...)
+	return nodeFilter
+}
+
+// DeleteLeakedResources deletes all available network interfaces on the node at node termination
+func (e *NodeTerminationCleaner) DeleteLeakedResources() error {
```

Comment:
This seems like a copy of eni_cleanup. Can you check if you can reuse the function with some arguments for their specific counters and tagset?